terminus-ui/cohort-date-range/src/cohort-date-range.component.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  ViewEncapsulation,
+} from '@angular/core';
+import { FormBuilder } from '@angular/forms';
+
+/**
+ * Represents date cohort object that passed in
+ */
+export interface TsDateCohort {
+  display: string;
+  range: {
+    startDate: string | Date;
+    endDate: string | Date;
+  };
+}
+
+/**
+ * Event object emitted by TsCohortDateRange when there is date range change
+ */
+
+export class TsCohortDateRangeChanged {
+  constructor(
+    public start: string | Date,
+    public end: string | Date,
+    public source: TsCohortDateRangeComponent,
+  ) { }
+}
+
+/**
+ * This is the cohort-date-range UI Component
+ *
+ *
+ * @example
+ * <ts-cohort-date-range
+ *              [allowCustomDates]="true"
+ *              [cohorts]="myCohorts"
+ *              (cohortDateRangeChange)="myFunc($event)"
+ *              <ts-date-range></ts-date-range>
+ *              <ts-select></ts-select>
+ * ></ts-cohort-date-range>
+ *
+ * <example-url>https://getterminus.github.io/ui-demos-release/components/cohort-date-range</example-url>
+ */
+@Component({
+  selector: 'ts-cohort-date-range',
+  templateUrl: './cohort-date-range.component.html',
+  styleUrls: ['./cohort-date-range.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  exportAs: 'tsCohortDateRange',
```

Comment:
Just for consistency (it doesn't matter functionally) we should put our changeDetection, encapsulation and exportAs at the bottom of the declaration block